core/trino-main/src/main/java/io/trino/execution/StageStateMachine.java

```diff
@@ -77,13 +79,13 @@
     private final PlanFragment fragment;
     private final Map<PlanNodeId, TableInfo> tables;
     private final SplitSchedulerStats scheduledStats;
-
     private final StateMachine<StageState> stageState;
     private final StateMachine<Optional<StageInfo>> finalStageInfo;
     private final AtomicReference<ExecutionFailureInfo> failureCause = new AtomicReference<>();
 
     private final AtomicReference<DateTime> schedulingComplete = new AtomicReference<>();
     private final Distribution getSplitDistribution = new Distribution();
+    private final Map<PlanNodeId, Distribution> tableGetSplitDistribution = new ConcurrentHashMap<>();
```

Comment:
I'm wondering what do we gain by knowing the distribution of get-splits time per table scan instead of the simpler cumulative time taken for get-splits per table scan ?
One file listing operation might feed multiple batches of splits, so it's possible that the distribution will just capture a few high values and many low values.

Reply:
With distribution, we have information about total time needed for getting splits, count, and information if there are some outliers splits.